### DISCUSSION TITLE
Deleting blank nodes

### DISCUSSION THREAD

**Discussion by syscharles** [apache/jena#1986] [Q&A] [answered]
Hello, 
Is there a way to delete blank nodes?
Thank you in advance.

**Comment by afs** [comment, marked as answer]
Could you describe your use in moire detail? [You said](https://github.com/afs/rdf-delta/discussions/233) you have data with blank nodes.

Java code calling the model API can delete statements containing blank nodes.

In SPARQL, a DELETE would find the triples with blank nodes in order to delete those triples.

How the app would find them depends on the form of the data. If the blank nod subjects have an inverse functional property, (a key to look them up by) then that is one simple way. Otherwise it needs an identifying pattern.
